### PR TITLE
[FEAT] scenario configuration to generate events

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

**Review by nbwhite — COMMENTED** [review]
Looks like there's two scenario.json files, one capitalized (Scenario.json) and one not capitalized. What's the difference between the two? You may consider renaming one of the them to be more descriptive.

**Review by avincigu — APPROVED** [review]

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignScenario.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        }
+
+        int index = 0;
+        for(; index < mainBatch.size() && index < burstBatch.size(); index++) {
+            PropertyArray mainBatchPart = mainBatch.getArray(index);
+            PropertyArray burstBatchPart = burstBatch.getArray(index);
+            mainBatchPart.addAll(burstBatchPart);
+        }
+
+        if(index < burstBatch.size()) {
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignScenario.java:260]: This check should be covered by the for loop condition below.

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 2]
```diff
@@ -42,66 +46,135 @@ void initialize() {
...
-                output.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
+                events.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
                 parameters.put("type", BOOT_STATES.ready.toString());
-                output.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
+                events.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
             }
             else
-                output.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
-            publishEvents(output, streamName_, burst_, timeDelay_, jpathToTimestamp_, output_, zone_);
+                events.addAll(generateEvents(parameters, EVENT_TYPE.BOOT.toString().toLowerCase()).getAsArray());
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:62]: This is a little confusing at first glance, put this statement in a block to make it more clear.

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 3]
```diff
@@ -129,13 +202,108 @@ private PropertyDocument generateEvents(Map<String, String> parameters, String t
...
+    private void generateEventsForRepeatMode(PropertyMap events, Map<String, String> parameters,
+                                             PropertyMap scenarioParameters) throws PropertyNotExpectedType, SimulatorException {
+        String repeatModeScenario = scenarioParameters.getString("mode");
+
+        PropertyArray publishEvents = new PropertyArray();
+        if(repeatModeScenario.equals("burst"))
+            publishEvents = foreignScenario_.generateEventsForBurstMode(events, repeatModeScenario);
+
+        else if (repeatModeScenario.equals("group-burst"))
+            publishEvents = foreignScenario_.generateEventsForBurstMode(events, repeatModeScenario);
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:227]: Should this be generateEventsForGroupBurstMode ?

**Review by nbwhite — COMMENTED** [review]
A few comments, most of them are just nitpicks, but there is a functional question about GroupBurst vs. Burst

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignScenario.java (hunk 1 above)
Comment (sindhu-106) [reply eventsim/src/main/java/com/intel/dai/eventsim/ForeignScenario.java:260]: loop breaks when index reaches min of 2 array sizes, As adding everything to 1 array, pending in 2nd array needs o be added to 1st array

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java (hunk 3 above)
Comment (sindhu-106) [reply eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:227]: Good Catch Nathan :)

**Review by nbwhite — APPROVED** [review]

**Pull request merged**